docs/source/notes/multiprocessing.rst

```diff
@@ -20,22 +20,26 @@ memory and will only send a handle to another process.
 This allows to implement various training methods, like Hogwild, A3C, or any
 others that require asynchronous operation.
 
-Sharing CUDA tensors
---------------------
-
-Sharing CUDA tensors between processes is supported only in Python 3, using
-a ``spawn`` or ``forkserver`` start methods. :mod:`python:multiprocessing` in
-Python 2 can only create subprocesses using ``fork``, and it's not supported
-by the CUDA runtime.
+CUDA in multiprocessing
+-----------------------
 
-.. warning::
+The CUDA runtime does not support the ``fork`` start method. However,
+:mod:`python:multiprocessing` in Python 2 can only create subprocesses using
+``fork``. So Python 3 and either ``spawn`` or ``forkserver`` start method are
+required to use CUDA in subprocesses.
 
-    CUDA API requires that the allocation exported to other processes remains
-    valid as long as it's used by them. You should be careful and ensure that
-    CUDA tensors you shared don't go out of scope as long as it's necessary.
-    This shouldn't be a problem for sharing model parameters, but passing other
-    kinds of data should be done with care. Note that this restriction doesn't
-    apply to shared CPU memory.
+.. note::
+  The start method can be set via either creating a context with
+  ``multiprocessing.get_context(...)`` or directly using
+  ``multiprocessing.set_start_method(...)``.
+
+Unlike CPU tensors, the sending process is required to keep the original tensor
+as long as the receiving process retains a copy of the tensor. It is implemented
+under the hood but requires users to follow the best practices for the program
+to run correctly. For example, the sending process must stay alive as long as
+the consumer process has references to the tensor, and the refcounting can not
```

Comment:
> For example, the sending process must stay alive as long as
the consumer process has references to the tensor.

Not sure if it obvious or worth mentioning, but sending process must stay alive as long as there are non zero elements in the 'outgoing' queue.

Reply:
Oh interesting. What will happen when there are shared CUDA tensor in queues but the sending process exits? Are those not freed by CUDA?

Reply:
No, as it is consumer's code which is doing ref-dec. In theory we can add this corner case, but it will complicate things by a lot.

Reply:
I see. But if the consumer either (1) eventually dies, or (2) gets the tensor from the queue and it gets GC'ed. The memory is still freed, right?